File: app/src/main/java/com/lumiyaviewer/lumiya/media/AudioIntentReceiver.java
package com.lumiyaviewer.lumiya.media;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import com.lumiyaviewer.lumiya.StreamingMediaService;

public class AudioIntentReceiver extends BroadcastReceiver {
    @Override
    public void onReceive(Context context, Intent intent) {
        if (intent.getAction().equals("android.media.AUDIO_BECOMING_NOISY")) {
            Intent intent2 = new Intent(context, (Class<?>) StreamingMediaService.class);
            intent2.setAction("com.lumiyaviewer.lumiya.ACTION_STOP_MEDIA");
            context.startService(intent2);
        }
    }
}

File: app/src/main/java/com/lumiyaviewer/lumiya/media/AudioIntentReceiver.kt
package com.lumiyaviewer.lumiya.media

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import com.lumiyaviewer.lumiya.StreamingMediaService

open class AudioIntentReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        if (intent.action == "android.media.AUDIO_BECOMING_NOISY") {
            context.startService(Intent(context, StreamingMediaService::class.java).apply {
                action = "com.lumiyaviewer.lumiya.ACTION_STOP_MEDIA"
            })
        }
    }
}
